api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java

```diff
@@ -92,6 +98,48 @@ public synchronized Order saveOrder(Order order, OrderContext orderContext) thro
 	}
 
 	/**
+	 * @see org.openmrs.api.OrderService#saveOrders(Set)
+	 */
+	public synchronized Set<Order> saveOrders(Set<Order> orders) {
+		Set<OrderGroup> orderGroups = new LinkedHashSet<OrderGroup>();
+
+		for (Order o : orders) {
+			if (o.getOrderId() == null) {
+				if (o.getOrderGroup() != null) {
+					o.setOrderGroup(getOrderGroup(orderGroups, o.getOrderGroup()));
+					setSortWeightForOrders(o);
+				}
+				saveOrder(o, null);
```

Comment:
This should call Context.getOrderService().saveOrder so that any advice around the method is not bypassed